src/main/scala/is/hail/sparkextras/OrderedRDD.scala

```diff
@@ -338,9 +338,35 @@ class OrderedRDD[PK, K, V] private(rdd: RDD[(K, V)], val orderedPartitioner: Ord
 
   override def getPreferredLocations(split: Partition): Seq[String] = rdd.preferredLocations(split)
 
-  def orderedLeftJoinDistinct[V2](other: OrderedRDD[PK, K, V2]): RDD[(K, (V, Option[V2]))] =
+  def groupByKey(): OrderedRDD[PK, K, Array[V]] = {
```

Comment:
Why don't you use this inside orderedLeftJoin?  Then it is just groupByKey followed by orderedLeftJoinDistinct.  Much simpler.